qemu/tests/migration_with_json_backing.py

```diff
@@ -0,0 +1,134 @@
+
+    def src_sn_chain_setup(images_tag):
+        params["image_name_%s" % images_tag[0]] = params["image_name"]
+        params["image_format_%s" % images_tag[0]] = params["image_format"]
+        images = get_img_objs(images_tag)
+        for image in images[1:-2]:
+            _create_image_with_backing(image, json_format=True)
+        return images
+
+    def blockdev_add_image(tag, is_backing_null=False):
+        image_params = params.object_params(tag)
+        devices = vm.devices.images_define_by_params(tag,
+                                                     image_params,
+                                                     'disk')
+        devices.pop()
+        for dev in devices:
+            if vm.devices.get_by_qid(dev.get_qid()):
+                continue
+            if isinstance(dev, QBlockdevFormatNode) and is_backing_null:
+                dev.params["backing"] = None
+            ret = vm.devices.simple_hotplug(dev, vm.monitor)
+            if not ret[1]:
+                test.fail("Failed to hotplug '%s': %s."
+                          % (dev, ret[0]))
+
+    def create_snapshot():
+        options = ["node", "overlay"]
+        cmd = "blockdev-snapshot"
+        arguments = params.copy_from_keys(options)
+        arguments.setdefault("node", "drive_%s" % params["base_tag"])
+        arguments.setdefault("overlay", "drive_%s" % params["snapshot_tag"])
+        return vm.monitor.cmd(cmd, dict(arguments))
+
+    def verify_backing_chain(info):
+        """Verify image's backing chain."""
+        for image, img_info in zip(images, reversed(info)):
+            base_image = None
+            if image.base_tag:
+                base_params = params.object_params(image.base_tag)
+                base_image = get_image_repr(image.base_tag,
+                                            base_params,
+                                            get_data_dir())
+            backing_info = img_info.get("full-backing-filename")
+            if backing_info and "json" in backing_info:
+                back_info = backing_info.strip("json:")
+                backing_info = json.loads(back_info)["file"]["filename"]
+            if base_image != backing_info:
+                test.fail(("backing chain check for image %s failed, backing"
+                           " file from info is %s, which should be %s.") %
+                          (image.image_filename, backing_info,
+                           base_image))
+
+    def check_backing_file(image):
+        out = json.loads(image.info(force_share=True, output="json"))
+        verify_backing_chain(out)
+
+    images_tag = params.get("image_chain").split()
+    images = src_sn_chain_setup(images_tag)
+    vm = env.get_vm(params["main_vm"])
+    vm.params["images"] = images_tag[-2]
+    vm.create()
+    for item in vm.devices.temporary_image_snapshots:
+        if images_tag[-2] in item:
+            vm.devices.temporary_image_snapshots.remove(item)
+            break
```

Comment:
It looks my vt patch works, but let's keep this for debug use, I will send out my patch for review asap